### PR TITLE
Refactor collection datastore to make its usage in tests easier

### PR DESCRIPTION
**Pull request by md2119**
## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
Unit

### CODE REVIEW

**Review by ksurabhi91 — COMMENTED** [review]
Some unit tests are failing because addcollection now returns 2 values and test returns one for some mocks

**Review by ksurabhi91 — COMMENTED** [review]
this pr https://github.com/stackrox/stackrox/pull/7340  also adds a method that will return collection id . I can make changes later to just use this new addcollection in unitests since it is  better

File: central/resourcecollection/datastore/datastore.go [hunk 1]
```diff
@@ -24,7 +27,7 @@ type DataStore interface {
 	GetMany(ctx context.Context, id []string) ([]*storage.ResourceCollection, error)
 
 	// AddCollection adds the given collection object and populates the `Id` field on the object
```
Comment (charmik-redhat) [inline central/resourcecollection/datastore/datastore.go:29]: "and returns the generated collection ID"

**Review by charmik-redhat — APPROVED** [review]
One minor comment. Otherwise looks good.

**Pull request merged**